sensors/imu/lsm9dsxx.c

```diff
@@ -252,23 +253,58 @@ static int lsm9dsxx_alloc(sensor_info_t *info, const char *args)
 	info->types = SENSOR_TYPE_ACCEL | SENSOR_TYPE_GYRO;
 
 	ctx->spiCtx.mode = SPI_MODE3;
-	ctx->spiCtx.speed = 6250000;
+	ctx->spiCtx.speed = 10000000;
+	ctx->spiSS = NULL;
+
+	ss = strchr(args, ':');
+	if (ss != NULL) {
+		ctx->spiSS = malloc(sizeof(oid_t));
+		if (ctx->spiSS == NULL) {
+			free(ctx);
+			return -ENOMEM;
+		}
+		*(ss++) = '\0';
+
+		/* get SPI SS oid */
+		ntries = 10;
+		while (lookup(ss, NULL, ctx->spiSS) < 0) {
+			ntries--;
+			if (ntries == 0) {
+				printf("lsm9dsxx: Can't open SPI SS pin\n");
+				free(ctx->spiSS);
+				free(ctx);
+				return -ETIMEDOUT;
+			}
+			usleep(10 * 1000);
+		}
+
+		/* configure SPI SS pin */
+		err = sensorsspi_initSS(ss);
+		if (err < 0) {
+			printf("lsm9dsxx: Can't configure SPI SS pin\n");
+			free(ctx->spiSS);
+			free(ctx);
+			return err;
+		}
+	}
```

Comment:
This could be (and perhaps should be) done inside of SPI lib. We wanted to hide this detail of SPI CS being driven via gpio driver from users.

Reply:
Done. I've renamed `sensorsspi_initSS()` to `sensorsspi_initDev()` and moved this code there.